### PR TITLE
Fixed stress test pipeline and other issues.

### PR DESCRIPTION
**Pull request by ManickaP**
Addresses multiple issues with HTTP stress:
- HTTP 1.1 won't run at all if HTTP 2.0 failed
  - Adding conditions that ensure that the step won't run only if build failed, not the previous stress run step.
- Pipeline naming inconsistencies
- Parameter nullability inspired by https://github.com/dotnet/runtime/pull/41321#pullrequestreview-477694852
  - Leaving `[DisallowNull]` in since that's the way it is in the interface: https://github.com/dotnet/runtime/blob/master/src/libraries/System.Private.CoreLib/src/System/Collections/Generic/IEqualityComparer.cs#L14
- The final report will log if the failed run has been intentionally cancelled, to improve investigation of existing H/2 failures.

### CODE REVIEW

File: eng/pipelines/libraries/stress/http.yml [hunk 1]
```diff
@@ -40,6 +40,7 @@ jobs:
     displayName: Build CLR and Libraries
   
   - bash: |
+      exit 1
```
Comment (wfurt) [inline eng/pipelines/libraries/stress/http.yml:43]: ???

Comment (ManickaP) [reply eng/pipelines/libraries/stress/http.yml:43]: This should be WIP right now, I'm testing something Santi suggested.

**Comment by ManickaP** [comment]
Runtime test failure: #41078,

**Review by alnikola — APPROVED** [review]
LGTM

**Pull request merged**